base/stream.jl

```diff
@@ -377,7 +377,7 @@ if OS_HANDLE != RawFD
 end
 
 function isopen(x::Union{LibuvStream, LibuvServer})
-    if x.status == StatusUninit || x.status == StatusInit
+    if x.status == StatusUninit || x.status == StatusInit || x.handle === C_NULL
```

Comment:
It looks like `isopen` is used outside the lock, e.g.,

https://github.com/JuliaLang/julia/blob/82741d6a99d86a9e1f85833e60b17e1cb298bf8d/base/stream.jl#L392-L398

but `x.handle` is not atomic. Actually, it looks like there are a couple of instances where `handle::Ptr{Cvoid}` does not have `@atomic`

```console
$ git grep -B1 -E '^    handle::Ptr\{Cvoid\}' base/ stdlib/Sockets
base/iostream.jl-mutable struct IOStream <: IO
base/iostream.jl:    handle::Ptr{Cvoid}
--
base/process.jl-    cmd::Cmd
base/process.jl:    handle::Ptr{Cvoid}
--
base/stream.jl-mutable struct PipeEndpoint <: LibuvStream
base/stream.jl:    handle::Ptr{Cvoid}
--
base/stream.jl-mutable struct TTY <: LibuvStream
base/stream.jl:    handle::Ptr{Cvoid}
--
stdlib/Sockets/src/PipeServer.jl-mutable struct PipeServer <: LibuvServer
stdlib/Sockets/src/PipeServer.jl:    handle::Ptr{Cvoid}
--
stdlib/Sockets/src/Sockets.jl-mutable struct TCPSocket <: LibuvStream
stdlib/Sockets/src/Sockets.jl:    handle::Ptr{Cvoid}
--
stdlib/Sockets/src/Sockets.jl-mutable struct TCPServer <: LibuvServer
stdlib/Sockets/src/Sockets.jl:    handle::Ptr{Cvoid}
--
stdlib/Sockets/src/Sockets.jl-mutable struct UDPSocket <: LibuvStream
stdlib/Sockets/src/Sockets.jl:    handle::Ptr{Cvoid}
```

Is this OK?

Reply:
The purpose of this PR is to make `x.handle` now constant data in many cases. The addition here is now more of a sanity check (against accidentally serializing the object).